packages/connect/src/types/api/ethereum/index.ts

```diff
@@ -74,10 +90,27 @@ export interface EthereumSignTypedDataMessage<T extends EthereumSignTypedDataTyp
         version?: string;
         chainId?: number | bigint;
         verifyingContract?: string;
-        salt?: ArrayBuffer;
+        salt?: ArrayBuffer | string;
     };
     message: { [fieldName: string]: any };
 }
+export const EthereumSignTypedDataMessage = Type.Object({
+    types: EthereumSignTypedDataTypes,
+    primaryType: Type.String(),
+    domain: Type.Object({
+        name: Type.Optional(Type.String()),
+        version: Type.Optional(Type.String()),
+        chainId: Type.Optional(Type.Union([Type.Number(), Type.BigInt()])),
+        verifyingContract: Type.Optional(Type.String()),
+        salt: Type.Optional(Type.Union([Type.ArrayBuffer(), Type.String()])),
+    }),
+    message: Type.Object(
+        {},
+        {
+            additionalProperties: Type.Any(),
+        },
+    ),
+});
```

Comment:
just a question, on line 97 we are not infering the EthereumSignTypedDataMessage interface. it is because it is generic right?